src/Client.php

```diff
@@ -286,4 +286,32 @@ protected function configure(array $options): void
 
         parent::configure($options);
     }
+
+    /**
+     * Waits for data to become available on the socket.
+     *
+     * @param float $maxSeconds the maximum amount of time to wait for data, in seconds
+     *
+     * @return bool returns true if data is available, false if the wait timed out
+     */
+    public function waitForData(float $maxSeconds): bool
+    {
+        $read = [$this->socket->getResource()];
+        $write = null;
+        $except = null;
+        $seconds = (int) \floor($maxSeconds);
+        $microseconds = (int) (($maxSeconds - $seconds) * 1e6);
+        $result = \stream_select($read, $write, $except, $seconds, $microseconds);
+        if (false === $result) {
+            $errorcode = \socket_last_error($this->socket->getResource());
+            $errormsg = \socket_strerror($errorcode);
+            throw new \RuntimeException("socket_select error $errorcode: $errormsg");
```

Comment:
Not sure this is the right exception to throw here.

Reply:
See https://github.com/chrome-php/wrench/blob/2cf557fad6b5fe6a1a01e18ad67c48eb889d216a/src/ConnectionManager.php#L98 for how error handling happens in other places.

Reply:
revised error handling: 799b8e0447bd1fa178e9e7030e1b0c932cd6f305